Tests fail to compile: error: reinterpret_cast from 'std::nullptr_t' to 'const char *' is not allowed

```
OsiTestSolverInterfaceIO.cpp:425:8: error: reinterpret_cast from 'std::nullptr_t' to 'const char *' is not allowed
  425 |                      reinterpret_cast<const char *> (NULL) /*integrality*/,
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
OsiTestSolverInterfaceIO.cpp:427:8: error: reinterpret_cast from 'std::nullptr_t' to 'const char **' is not allowed
  427 |                      reinterpret_cast<const char **> (NULL) /*colnam*/, 
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
OsiTestSolverInterfaceIO.cpp:428:8: error: reinterpret_cast from 'std::nullptr_t' to 'const char **' is not allowed
  428 |                      reinterpret_cast<const char **> (NULL) /*rownam*/);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
```

NULL should be replaced with nullptr in C++ code.

clang-18
FreeBSD 14.0

--- a/test/OsiTestSolverInterfaceIO.cpp
+++ b/test/OsiTestSolverInterfaceIO.cpp
@@ -422,10 +422,10 @@ OsiTestSolverInterface::writeMps(const char *filename,
    writer.setMpsData(*getMatrixByCol(), getInfinity(),
 		     getColLower(), getColUpper(),
 		     getObjCoefficients(), 
-		     reinterpret_cast<const char *> (NULL) /*integrality*/,
+		     NULL /*integrality*/,
 		     getRowLower(), getRowUpper(),
-		     reinterpret_cast<const char **> (NULL) /*colnam*/, 
-		     reinterpret_cast<const char **> (NULL) /*rownam*/);
+		     NULL /*colnam*/,
+		     NULL /*rownam*/);
    std::string fname = filename;
    if (extension)
    { if (extension[0] != '\0' && extension[0] != '.')

